packages/react-html/README.md

```diff
@@ -12,112 +12,267 @@ $ yarn add @shopify/react-html
 
-The `<Script />` component lets you render `<script>` tags in your document dynamically as part of your react app.
+The `<Script />` component lets you render `<script>` tags in your document dynamically as part of your react app. It supports all the props of a basic `script` tag.
+
+```tsx
+import {Script} from '@shopify/react-html';
 
-```ts
 <Script
   src="./some-script.js"
   integrity="some-integrity-hash"
   crossOrigin="anonymous"
+/>;
+```
+
+### Server
+
+The server entrypoint allows you to easily construct an HTML document for the response. It also re-exports all of the components noted above.
+
+#### `<Html />` and `render()`
+
+The `<Html>` component serves as a top level wrapper for a React application, allowing you to avoid needing any kind of server-side template. The `render()` function creates a stringified version of the HTML document with an appropriate DOCTYPE.
+
+```tsx
+import * as React from 'react';
+import {render, Html} from '@shopify/react-html/server';
+
+import App from '../app';
+
+export default function middleware(ctx, next) {
+  ctx.body = render(
+    <Html>
+      <App />
+    </Html>,
+  );
+
+  await next();
+}
+```
+
+The component will automatically propagate any usage of the [`react-helmet` library](https://github.com/nfl/react-helmet) in your app’s content to manipulate the title or other top level HTML or HEAD attributes.
+
+##### Props
+
+The `<Html />` component accepts the following props:
+
+- `manager`: a `Manager` instance. When provided, the `Html` component will extract all the information from this object and place it in an appropriate place in the document.
+- `children`: the application. It will be rendered to a string and placed inside a `div` with an ID of `app`.
+- `locale`: the language to use for the HTML `lang` attribute.
+- `styles`: descriptors for any style tags you want to include in the HEAD of the document.
+- `scripts`: descriptors for any script tags you want to include in your document. All scripts passed to this property will be deferred by appending them to the end of the document. We encourage this as a default because it improves the initial rendering performance of your page.
+- `blockingScripts`: descriptors for any script tags you want to include in the HEAD of the document. These will block HTML parsing until they are evaluated, so use them carefully.
+- `headMarkup`: additional JSX to be embedded in the head of the document (after styles, but before blocking scripts).
+- `bodyMarkup`: additional JSX to be embedded in the body of the document (before serialization markup and deferred scripts).
+
+#### `<Serialize />`
+
+The Serialize component is a component that will take care of rendering a `script` tag with a serialized version of the `data` prop. You should not need this directly; prefer using the `createSerializer()` function within your app code instead.
+
+## Migrating from `@shopify/react-html@4.x`/ `@shopify/react-serialize@1.x` to `@shopify/react-html@5.x`
```

Comment:
I love migration notes, but I think they should be in a separate file. sewing-kit recently added a [migration-guide.md](https://github.com/Shopify/sewing-kit/blob/master/migration-guide.md) should we follow that?

Reply:
Agreed that this belongs in a separate migration doc